Tests/Unit/LoRaWan/DevAddrTests.cs

```diff
@@ -3,13 +3,40 @@
 
 namespace LoRaWan.Tests.Unit
 {
+    using System;
     using LoRaWan;
+    using LoRaWan.Tests.Common;
     using Xunit;
 
     public class DevAddrTests
     {
         private readonly DevAddr subject = new(0xeb6f7bde);
 
+        [Theory]
+        [InlineData(uint.MaxValue)]
+        [InlineData(uint.MinValue)]
+        [InlineData(0xeb6f7bde)]
+        public void Init(uint value)
+        {
+            // arrange
+            var maxValue = new DevAddr(value);
+
+            // act
+            var result = new DevAddr(maxValue.NetworkId, maxValue.NetworkAddress);
+
+            // assert
+            Assert.Equal(maxValue, result);
+        }
+
+        [Fact]
+        public void Init_Validates_Limits()
+        {
+            var maxValue = new DevAddr(uint.MaxValue);
+
+            Assert.Throws<ArgumentException>(() => new DevAddr(maxValue.NetworkId + 1, maxValue.NetworkAddress));
```

Comment:
I don't think the bounds being checked here are correct for each part of the `DevAddr`. Also, we don't really assert which of the two arguments failed.